src/audio/dcblock/dcblock_generic.c

```diff
@@ -40,27 +40,36 @@ static void dcblock_s16_default(const struct comp_dev *dev,
 {
 	struct comp_data *cd = comp_get_drvdata(dev);
 	struct dcblock_state *state;
-	int16_t *x;
-	int16_t *y;
+	int16_t *x = source->r_ptr;
+	int16_t *y = sink->w_ptr;
 	int32_t R;
 	int32_t tmp;
 	int idx;
 	int ch;
-	int i;
+	int i, n, nmax;
 	int nch = source->channels;
+	int samples = nch * frames;
 
-	for (ch = 0; ch < nch; ch++) {
-		state = &cd->state[ch];
-		R = cd->R_coeffs[ch];
-		idx = ch;
-		for (i = 0; i < frames; i++) {
-			x = audio_stream_read_frag_s16(source, idx);
-			y = audio_stream_write_frag_s16(sink, idx);
-			tmp = dcblock_generic(state, R, *x << 16);
-			*y = sat_int16(Q_SHIFT_RND(tmp, 31, 15));
-			idx += nch;
+	while (samples) {
+		nmax = audio_stream_bytes_without_wrap(source, x) >> 1;
```

Comment:
There's a nicer function for this purpose, see audio_stream_samples_without_wrap_s16/s24/s32()

Reply:
Thanks